File: semelhanca_atomica - Elias GSI10AM.py

import os
os.system("cls")    
def semelhanca_atomica():
    prot1 = int(input("Número de protões do 1° elemento: "))
    neut1 = int(input("Número de neutrões do 1° elemento: "))
    prot2 = int(input("Número de protões do 2° elemento: "))
    neut2 = int(input("Número de neutrões do 2° elemento: "))

    if prot1 == prot2 and neut1 != neut2:
        print("Os elementos são isótopos.")
    elif (prot1 + neut1) == (prot2 + neut2):
        print("Os elementos são isóbaros.")
    elif neut1 == neut2 and prot1 != prot2:
        print("Os elementos são isótonos.")
    else:
        print("Os elementos não são isótopo, isóbaro ou isótono.")

def calcular_particulas():
    Z = int(input("Digite o número atômico (Z): "))
    A = int(input("Digite o número de massa (A): "))
    C = int(input("Digite a carga do átomo (C): "))
    print('')
    P = Z
    N = A - Z
    E = Z - C

    print("Número de protões:", P)
    print("Número de neutrões:", N)
    print("Número de eletrões:", E)
    
    print("")
while True :
    print("======== Projecto de TLP ========")
    print("Nome:Elias Timóteo Manuel   Nº14\nTurma:GSI10AM")
    print("=================================")
    print("Escolha uma opção:")
    print("1 - Semelhança Atómica")
    print("2 - Calcular partículas")
    print("3 - Sair")
    print('=================================')
    op = int(input("Opção: "))
    os.system("cls")
    if op == 1:
        semelhanca_atomica()
    elif op == 2:
        calcular_particulas()
    elif op == 3:
        print("Programa finalizado!")
        break
    else:
        print("Opção incorreta!")

File: semelhanca_atomica - Elias GSI10AM.c
#include<stdio.h>
#include<locale.h>
#include<stdlib.h>

 int main()
{
    setlocale(LC_ALL,"Portuguese");
	int Z, A, P, N, C, j, op ,prot1 ,prot2 ,neut1,neut2;

    printf("\t===================Projecto de TLP====================\t\n");
	printf("==========================\n");
	printf("Nome: Elias Manuel\n");
	printf("Turma: GSI10AM\nNº14\n");
    do{
        printf("*==========================*\n");
        printf("Escolha uma opção: \n");
        printf("1 - Semelhança atómica\n");
        printf("2 - Calcular particulas\n");
        printf("3 - Sair\n");
        scanf("%d",&op);
        printf("================================\n");

        switch(op){
            case 1:
            printf("Insira o número de protões do 1º elemento: ");
            scanf("%d",&prot1);
            printf("Insira o número de neutrões do 1º elemento:");
            scanf("%d",&neut1);
            printf("Insira o número de protões do 2º elemento: ");
            scanf("%d",&prot2);
            printf("Insira o número de neutrões do 2º elemento:");
            scanf("%d",&neut2);
            printf("\n");
            if (prot1==prot2 && neut1!=neut2){

                printf("R:Os elementos são Isótopos.\n");
            }else if (prot1!=prot2 && neut1==neut2){
                printf("\n");
                printf("R:Os elementos são Isótonos.\n");
            }else if ((prot1+neut1)==(prot2+neut2)){
                printf("\n");
                printf("R:Os elementos são Isóbaros.\n");
            }else{
                printf("\n");
                printf("R:Os elementos não são Isótopos, Isótonos nem Isóbaros!\n");
            }
            break;

            case 2:
            printf("Digite o número de atómico (Z):");
            scanf("%d",&Z);
            printf("Digite o número de massa (A):");
            scanf("%d",&A);
            printf("Digite a carga do átomo (C):");
            scanf("%d",&C);

            P = Z ;
            N = A - Z;
            j = Z - C;
            printf("\n");
            printf("Número de protões:%d\n",P);
            printf("Número de neutrões:%d\n",N);
            printf("Número de eletrões:%d \n",j);
            break;
            case 3:
            printf("Programa encerrado!\n");
            break;
            default:
            printf("Opção inválida!\n");
        }
        printf("\n");
    }	while(op!=3);
        return 0;
}
